src/main/java/jdbc/dao/TicketDAOImpl.java

```diff
@@ -0,0 +1,111 @@
+package jdbc.dao;
+
+import homework.TicketType;
+import jdbc.pojo.Ticket;
+import jdbc.util.ConnectionManager;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TicketDAOImpl implements TicketDAO {
+    private final static TicketDAOImpl INSTANCE = new TicketDAOImpl();
+    private final static String SAVE_SQL = """
+            INSERT INTO ticket
+             (user_id, ticket_type)
+             VALUES(?,?)
+            """;
+    private final static String GET_BY_ID_SQL = """
+            SELECT * FROM ticket
+            WHERE id=?
+            """;
+    private final static String GET_BY_USER_ID_SQL = """
+            SELECT * FROM ticket
+            WHERE user_id=?
+            """;
+    private static final String UPDATE_TYPE_SQL = """
+            UPDATE ticket SET ticket_type=?
+            WHERE id=?
+            """;
+
+
+    public Ticket save(Ticket t) {
+        try (var connection = ConnectionManager.open();
+             var statement = connection.prepareStatement(SAVE_SQL)) {
+            statement.setInt(1, t.getUserId());
+            statement.setString(2, t.getTicketType().toString());
+
+            statement.executeUpdate();
+            return t;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Ticket getTicket(int id) {
+        try (var connection = ConnectionManager.open();
+             var statement = connection.prepareStatement(GET_BY_ID_SQL)) {
+            statement.setInt(1, id);
+            try (var result = statement.executeQuery()) {
+                Ticket ticket = null;
+                if (result.next()) {
+                    ticket = buildTicket(result);
+                    return ticket;
+                }
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+
+        }
+        return null;
+    }
+
+    private static Ticket buildTicket(ResultSet result) throws SQLException {
```

Comment:
move after public methods, and why its static?

Reply:
I extracted this with idea/ it made method static

Reply:
just check it as well